sky/resources.py

```diff
@@ -1327,8 +1344,8 @@ def _try_validate_volumes(self) -> None:
                 # Refer to https://cloud.google.com/compute/docs/disks/local-ssd#machine-series-lssd # pylint: disable=line-too-long
                 self.cloud.check_disk_tier_enabled(self.instance_type,
                                                    volume['disk_tier'])
-            if (need_region_or_zone and self._region is None and
-                    self._zone is None):
+            if (not kubernetes and need_region_or_zone and
+                    self._region is None and self._zone is None):
```

Comment:
curious - how is failover across contexts handled? E.g., if a PVC exists in contextA and the job fails over to contextB, will we throw an error?